File: ReceiptHealth/add-items.py
import http.client
import json

def make_request(method, path, body=None):
    conn = http.client.HTTPConnection('localhost', 5002)
    headers = {'Content-Type': 'application/json'} if body else {}
    
    body_data = json.dumps(body) if body else None
    conn.request(method, path, body_data, headers)
    
    response = conn.getresponse()
    data = response.read().decode()
    
    try:
        return response.status, json.loads(data)
    except:
        return response.status, data

# Get shopping lists
print('📋 Fetching shopping lists...')
status, lists = make_request('GET', '/api/shopping-lists')

if not lists:
    # Create a new list
    print('➕ Creating new shopping list...')
    status, new_list = make_request('POST', '/api/shopping-lists', {'name': 'My Shopping List'})
    list_id = new_list['id']
    print(f'✅ Created list with ID: {list_id}')
else:
    # Use the first list
    list_id = lists[0]['id']
    print(f'✅ Using existing list ID: {list_id}')

# Items to add
items = ['banana', 'kefir', 'mug', 'potato', 'onion']

print(f'\n🛒 Adding {len(items)} items to shopping list...')
for item_name in items:
    status, response = make_request('POST', f'/api/shopping-lists/{list_id}/items', {
        'itemName': item_name,
        'quantity': 1
    })
    
    if status == 201:
        print(f'✅ Added: {item_name}')
    else:
        print(f'❌ Failed to add {item_name}: {response}')

# Get the updated list
print('\n📋 Final shopping list:')
status, final_list = make_request('GET', f'/api/shopping-lists/{list_id}')
print(json.dumps(final_list, indent=2))

File: ReceiptHealth/add-items.js
const http = require('http');

async function makeRequest(method, path, body = null) {
    return new Promise((resolve, reject) => {
        const options = {
            hostname: 'localhost',
            port: 5100,
            path: path,
            method: method,
            headers: body ? {
                'Content-Type': 'application/json',
                'Content-Length': Buffer.byteLength(JSON.stringify(body))
            } : {}
        };

        const req = http.request(options, (res) => {
            let data = '';
            res.on('data', (chunk) => data += chunk);
            res.on('end', () => {
                try {
                    resolve({ status: res.statusCode, data: JSON.parse(data) });
                } catch (e) {
                    resolve({ status: res.statusCode, data: data });
                }
            });
        });

        req.on('error', reject);
        
        if (body) {
            req.write(JSON.stringify(body));
        }
        req.end();
    });
}

async function addItems() {
    try {
        // Get shopping lists
        console.log('📋 Fetching shopping lists...');
        const listsResponse = await makeRequest('GET', '/api/shopping-lists');
        
        let listId;
        if (listsResponse.data.length === 0) {
            // Create a new list
            console.log('➕ Creating new shopping list...');
            const createResponse = await makeRequest('POST', '/api/shopping-lists', {
                name: 'My Shopping List'
            });
            listId = createResponse.data.id;
            console.log(`✅ Created list with ID: ${listId}`);
        } else {
            // Use the first list
            listId = listsResponse.data[0].id;
            console.log(`✅ Using existing list ID: ${listId}`);
        }

        // Items to add
        const items = ['banana', 'kefir', 'mug', 'potato', 'onion'];
        
        console.log(`\n🛒 Adding ${items.length} items to shopping list...`);
        for (const itemName of items) {
            const response = await makeRequest('POST', `/api/shopping-lists/${listId}/items`, {
                itemName: itemName,
                quantity: 1
            });
            
            if (response.status === 201) {
                console.log(`✅ Added: ${itemName}`);
            } else {
                console.log(`❌ Failed to add ${itemName}: ${JSON.stringify(response.data)}`);
            }
        }

        // Get the updated list
        console.log('\n📋 Final shopping list:');
        const finalResponse = await makeRequest('GET', `/api/shopping-lists/${listId}`);
        console.log(JSON.stringify(finalResponse.data, null, 2));

    } catch (error) {
        console.error('❌ Error:', error.message);
    }
}

addItems();
